### ISSUE TITLE
TRUNCATE waits for offline replicas.

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#31989]
Scenario  - 3 replicas cluster. 
```
DROP database IF EXISTS test ON CLUSTER 'testcluster';
create database test ON CLUSTER 'testcluster';

CREATE TABLE IF NOT EXISTS test.pets ON CLUSTER 'testcluster' (id UInt32, name String, eventTime DateTime) Engine = ReplicatedMergeTree('/clickhouse/iond_shard2/test/tables/pets/{shard}', '{replica}') PARTITION BY toYYYYMMDD(eventTime) ORDER BY (id,eventTime);

insert into test.pets (id, name, eventTime) values(1, 'coco', '2021-04-26 18:07:13');
```

after that shutdown one of replicas, and run TRUNCATE command on one of the active replicas.

Result - TRUNCATE hungs. 

Stacktrace: 

```
Poco::EventImpl::waitImpl(long)
DB::StorageReplicatedMergeTree::waitForTableReplicaToProcessLogEntry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::ReplicatedMergeTreeLogEntryData const&, bool)
DB::StorageReplicatedMergeTree::waitForAllTableReplicasToProcessLogEntry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::ReplicatedMergeTreeLogEntryData const&, bool)
DB::StorageReplicatedMergeTree::truncate(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::Context const&, DB::TableExclusiveLockHolder&)
DB::InterpreterDropQuery::executeToTableImpl(DB::ASTDropQuery const&, std::__1::shared_ptr<DB::IDatabase>&, StrongTypedef<DB::UInt128, DB::UUIDTag>&)
DB::InterpreterDropQuery::executeToTable(DB::ASTDropQuery const&)
DB::InterpreterDropQuery::execute()

DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool)
DB::TCPHandler::runImpl()
DB::TCPHandler::run()
Poco::Net::TCPServerConnection::start()
Poco::Net::TCPServerDispatcher::run()
Poco::PooledThread::run()
Poco::ThreadImpl::runnableEntry(void*)
start_thread
clone
```
 
Repeats with replication_alter_partitions_sync=0 / 1. 

Related
https://github.com/ClickHouse/ClickHouse/issues/27178 
https://github.com/ClickHouse/ClickHouse/pull/27931

**Comment by filimonov** [comment]
Reproduces on 21.8.11 and 21.3.17.

Does not reproduce on 21.11.

**Comment by tavplubix** [comment]
> Reproduces on 21.8.11 and 21.3.17.
> 
> Does not reproduce on 21.11.

Because it's fixed in #27931, i.e. in 21.10 and newer versions
